### ISSUE TITLE
Desktop file doesn't validate

### ISSUE THREAD

**Issue by Enchufa2** [jacktrip/jacktrip#705]
I'm updating jacktrip in Fedora to v1.6.2, but the desktop file doesn't [validate](https://docs.fedoraproject.org/en-US/packaging-guidelines/#_desktop_file_install_usage) anymore. I see:

```
+ desktop-file-validate /builddir/build/BUILDROOT/jacktrip-1.6.2-1.fc36.x86_64//usr/share/applications/org.jacktrip.JackTrip.desktop
/builddir/build/BUILDROOT/jacktrip-1.6.2-1.fc36.x86_64//usr/share/applications/org.jacktrip.JackTrip.desktop: error: value "/bin/bash -c 'if [ -z "$1" ]; then jacktrip; else jacktrip --gui --deeplink $1; fi' /bin/bash %u" for key "Exec" in group "Desktop Entry" contains a reserved character ''' outside of a quote
/builddir/build/BUILDROOT/jacktrip-1.6.2-1.fc36.x86_64//usr/share/applications/org.jacktrip.JackTrip.desktop: error: value "/bin/bash -c 'if [ -z "$1" ]; then jacktrip; else jacktrip --gui --deeplink $1; fi' /bin/bash %u" for key "Exec" in group "Desktop Entry" contains a non-escaped character '$' in a quote, but it should be escaped with two backslashes ("\\$")
/builddir/build/BUILDROOT/jacktrip-1.6.2-1.fc36.x86_64//usr/share/applications/org.jacktrip.JackTrip.desktop: error: value "/bin/bash -c 'if [ -z "$1" ]; then jacktrip; else jacktrip --gui --deeplink $1; fi' /bin/bash %u" for key "Exec" in group "Desktop Entry" contains a reserved character ';' outside of a quote
/builddir/build/BUILDROOT/jacktrip-1.6.2-1.fc36.x86_64//usr/share/applications/org.jacktrip.JackTrip.desktop: error: value "/bin/bash -c 'if [ -z "$1" ]; then jacktrip; else jacktrip --gui --deeplink $1; fi' /bin/bash %u" for key "Exec" in group "Desktop Entry" contains a reserved character ';' outside of a quote
/builddir/build/BUILDROOT/jacktrip-1.6.2-1.fc36.x86_64//usr/share/applications/org.jacktrip.JackTrip.desktop: error: value "/bin/bash -c 'if [ -z "$1" ]; then jacktrip; else jacktrip --gui --deeplink $1; fi' /bin/bash %u" for key "Exec" in group "Desktop Entry" contains a reserved character '$' outside of a quote
/builddir/build/BUILDROOT/jacktrip-1.6.2-1.fc36.x86_64//usr/share/applications/org.jacktrip.JackTrip.desktop: error: value "/bin/bash -c 'if [ -z "$1" ]; then jacktrip; else jacktrip --gui --deeplink $1; fi' /bin/bash %u" for key "Exec" in group "Desktop Entry" contains a reserved character ';' outside of a quote
/builddir/build/BUILDROOT/jacktrip-1.6.2-1.fc36.x86_64//usr/share/applications/org.jacktrip.JackTrip.desktop: error: value "/bin/bash -c 'if [ -z "$1" ]; then jacktrip; else jacktrip --gui --deeplink $1; fi' /bin/bash %u" for key "Exec" in group "Desktop Entry" contains a reserved character ''' outside of a quote
```

What's the proper fix for this? I could apply a temporary patch until this is solved in the next release.

**Comment by mattahorton** [comment]
Hm. I made this change. 

I can address this. We're already planning to release an update for a critical bug fix, and this should be included.

**Comment by Enchufa2** [comment]
Maybe running `desktop-file-validate` could be added to your CI workflow to avoid these issues in the future.

**Comment by mattahorton** [comment]
> validate

It definitely will be.

**Comment by ntonnaett** [comment]
Interesting that this didn't pop up with the flatpak build.

**Comment by mattahorton** [comment]
Agreed @ntonnaett. @Enchufa2 could you try this Exec line?

`Exec=/bin/bash -c "if [ -z ""\$"1" ]; then jacktrip; else jacktrip --gui --deeplink ""\$"1"; fi" /bin/bash %u`

That's the line I've used in #706 for the fix that should go in 1.6.3.

**Comment by Enchufa2** [comment]
I confirm that the fix works. Building v1.6.2 with the patch for Fedora 35, 36, 37 and rawhide.

**Comment by mattahorton** [comment]
Thanks for that. We'll get that in for 1.6.3 as well.

**Comment by mattahorton** [comment]
@Enchufa2 this fix is included in the 1.6.3 release we got out yesterday.

**Comment by Enchufa2** [comment]
Thanks, I saw the release this morning and sent new updates for v1.6.3.
